interfaces/builtin/mount_control.go

```diff
@@ -503,7 +574,8 @@ func (iface *mountControlInterface) AppArmorConnectedPlug(spec *apparmor.Specifi
 			typeRule = "fstype=(" + strings.Join(types, ",") + ")"
 		}
 
-		options := strings.Join(mountInfo.options, ",")
+		// only pass the kernel mount options on to apparmor
+		options := strings.Join(filterKernelMountOptions(mountInfo.options), ",")
```

Comment:
A very silly question (sorry!) what will happen with filesystem specific option like (for vfat) "uid=1000". Right now this is filtered out as this line only takes into account generic kernel options but not filesystem specific ones. Is this intentional? Or should it be possible to write: `uid=1000` here and see it in the apparmor spec?

A test for this could look like this I suppose (this fails right now due to the above):
```go
diff --git a/interfaces/builtin/mount_control_test.go b/interfaces/builtin/mount_control_test.go
index ef390159a4..68ea23e47d 100644
--- a/interfaces/builtin/mount_control_test.go
+++ b/interfaces/builtin/mount_control_test.go
@@ -74,6 +74,10 @@ plugs:
     where: $SNAP_COMMON/mnt/**
     type: [ext2, ext3, ext4]
     options: [nofail, sync, acl]
+  - what: /dev/sdc
+    where: $SNAP_COMMON/mnt/**
+    type: [vfat]
+    options: [uid=1000]
 apps:
  app:
   plugs: [mntctl]
@@ -354,6 +358,10 @@ func (s *MountControlInterfaceSuite) TestAppArmorSpec(c *C) {
        c.Assert(spec.SnippetForTag("snap.consumer.app"), testutil.Contains, expectedMountLine6)
        expectedUmountLine6 := `umount "/var/snap/consumer/common/mnt/**{,/}",`
        c.Assert(spec.SnippetForTag("snap.consumer.app"), testutil.Contains, expectedUmountLine6)
+
+       expectedMountLine7 := `mount fstype=(vfat) options=(uid=1000) ` +
+               `"/dev/sdc" -> "/var/snap/consumer/common/mnt/**{,/}",`
+       c.Assert(spec.SnippetForTag("snap.consumer.app"), testutil.Contains, expectedMountLine7)
 }
 

```

Reply:
Thanks for giving me the needed context here! This makes sense and I see you also added spread test code to validate this - thanks for this too.